dash/include/dash/Array.h

```diff
@@ -1014,7 +1066,10 @@ class Array
   void barrier() const
   {
     DASH_LOG_TRACE_VAR("Array.barrier()", m_team);
-    if (nullptr != m_team && dash::Team::Null() != *m_team) {
+    if (nullptr != m_globmem) {
+      m_globmem->flush_all();
+    }
+    if (nullptr != m_team && *m_team != dash::Team::Null()) {
```

Comment:
Can m_team ever be a nullptr? It seems to be initialized from a dash::Team reference in all c'tors.